tests/test_pyfunction.rs

```diff
@@ -185,7 +185,9 @@ fn test_from_py_with_defaults() {
     }
 
     #[pyfunction(signature = (len=0))]
-    fn from_py_with_default(#[pyo3(from_py_with = "PyAny::len")] len: usize) -> usize {
+    fn from_py_with_default(
+        #[pyo3(from_py_with = "<Bound<'_, _> as PyAnyMethods>::len")] len: usize,
```

Comment:
Does this work?

```suggestion
        #[pyo3(from_py_with = "PyAnyMethods::len")] len: usize,
```

Reply:
Sadly not:
```cannot infer type for fn pointer `fn(&_) -> Result<usize, pyo3::PyErr>```
(It also fails on main)

Reply:
Ah, frustrating. I wonder if that's because we cast to `fn(_)` for the GIL Refs compatibility. I think not a blocker for release, I'll add it to the list of possible follow up tasks in the tracking issue.

Reply:
> I wonder if that's because we cast to `fn(_)` for the GIL Refs compatibility

That would have been my guess as well, and that both variants take reference arguments does not help either I assume...